### PR TITLE
Rewording and provide new URL path `/run-test` instead of `/check` (former `/domain_check`)

### PR DESCRIPTION
**Pull request by pnax**
## Purpose

Take into consideration reviews and suggestions in meeting and partially from https://github.com/zonemaster/zonemaster-gui/pull/333#issuecomment-1249376427.
Mainly this is about rewording menu entry and everything around the "Domain check" words. This PR goes a little bit further with a code refactoring to reflect such changes.

This PR is to be merged before #333. Conflicts will be resolved in #333.

## Context

https://github.com/zonemaster/zonemaster-gui/pull/333 and meeting discussion
updates #334

## Changes

* Update wording and API `/run-test`, `/run-test/<domain>`, "Run test", "Run domain test".
* Refactor code to reflect such changes in variables, components and file names.
* Adapt e2e tests

## How to test this PR

Tests should pass.

### CODE REVIEW

**Review by mvw-afnic — DISMISSED** [review]
Looks good to me. Translations will need an update before next release.

File: src/assets/i18n/fr.json [hunk 1]
```diff
@@ -52,9 +52,10 @@
     "TOOLTIP_INFO": "Info - contient des informations qui peuvent être d'intérêt pour la personne lançant le test, ces messages ne sont pas des erreurs",
     "TOOLTIP_NOTICE": "Notice - destiné à informer la personne en charge de la zone, ne correspond pas nécessairement à un problème de configuration",
     "TOOLTIP_WARNING": "Warning - destiné aux erreurs détectées dans certains cas d'usage spécifiques",
+    "Test completed": null,
```
Comment (matsduf) [inline src/assets/i18n/fr.json:55]: Should this not be given a French translation?

File: src/assets/i18n/da.json [hunk 2]
```diff
@@ -52,9 +52,10 @@
...
     "TOOLTIP_NOTICE": "Notice - betyder noget, der skal være kendt af zonens administrator, men det er ikke nødvendigvis et problem.",
     "TOOLTIP_WARNING": "Warning - betyder noget, der under visse omstændigheder vil være et problem, men som sandsynligvis ikke vil blive bemærket af den almindelige bruger.",
+    "Test completed": null,
     "There is no delegation for the zone": "Der er ikke nogen delegation for zonen",
     "UNDELEGATED_PLURAL": "Ikke-delegerede",
     "UNDELEGATED_SINGULAR": "Ikke-delegeret",
     "What is an undelegated domain test?": "Hvad er en ikke-delegeret test?",
     "Zonemaster Backend is not available": "Zonemaster Backend er ikke tilgængelig"
-}
+}
```
Comment (matsduf) [inline src/assets/i18n/da.json:61]: How are the files created? Why is newline missing on last line. It is easier to handle the file if it has.

File: src/assets/i18n/fr.json (hunk 1 above)
Comment (pnax) [reply src/assets/i18n/fr.json]: good catch, I've added the translation

File: src/assets/i18n/da.json (hunk 2 above)
Comment (pnax) [reply src/assets/i18n/da.json:61]: with `npm run i18n:extract`

**Comment by matsduf** [comment]
> Looks good to me. Translations will need an update before next release.

Yes, when this is merge we should create an issue for translation.

File: docs/API.md [hunk 3]
```diff
@@ -8,17 +8,17 @@ A path is added to the Zonemaster GUI base URL, e.g.
 
 The supported paths are the following:
 
-* `check`: the default path, access the input form
-* `domain_check`: same as `check`, but *deprecated*
-* `check/<domain>`: populate the input form with `<domain>`, which is expected
+* `run-test`: the default path, access the input form
+* `domain_check`: same as `run-test`, but *deprecated*
+* `run-test/<domain>`: populate the input form with `<domain>`, which is expected
```
Comment (matsduf) [inline docs/API.md:13]: run-test/<domain> works, but domain_check/<domain> does not work.

image

File: src/assets/i18n/da.json (hunk 2 above)
Comment (matsduf) [reply src/assets/i18n/da.json:61]: Can that be made to add a newline to the files?

**Review by matsduf — CHANGES_REQUESTED** [review]

File: docs/API.md (hunk 3 above)
Comment (pnax) [reply docs/API.md:13]: This is the expected behavior (similar to what was done in #334 too).

Comment (matsduf) [reply docs/API.md:13]: I am not against the behavior, and the document could be read that way, but I suggest that it is clarified that the rewrite is only done when not followed by "/<domain>".

Comment (pnax) [reply docs/API.md:13]: I clarified this later in the document.

**Review by blacksponge — APPROVED** [review]
looks good to me

**Comment by pnax** [comment]
Rebased on top of develop to fix a conflict.

File: src/assets/i18n/en.json [hunk 4]
```diff
@@ -16,9 +16,7 @@
     "Digest type required": "Digest type required",
     "Disable IPv4": "Disable IPv4",
     "Disable IPv6": "Disable IPv6",
-    "Domain check": "Domain check",
-    "Domain checked completed": "Domain checked completed",
-    "Domain name": "Domain name",
+    "Domain name": null,
```
Comment (matsduf) [inline src/assets/i18n/en.json:19]: Is that really correct? I expected it to be
```
"Domain name": "Domain name",
```

File: src/assets/i18n/en.json [hunk 5]
```diff
@@ -43,6 +41,8 @@
     "Program versions": "Program versions",
     "Reset": "Reset",
     "Results from previous tests are not available": "Results from previous tests are not available",
+    "Run": null,
```
Comment (matsduf) [inline src/assets/i18n/en.json:44]: Is that really correct? I expected it to be
```
"Run": "Run",
```

**Review by matsduf — CHANGES_REQUESTED** [review]
Two small things, else it looks good.

**Review by tgreenx — DISMISSED** [review]
LGTM - tested and works as expected.

**Review by matsduf — DISMISSED** [review]

**Comment by pnax** [comment]
rebased on top of develop, please re-review

**Review by matsduf — DISMISSED** [review]

**Review by tgreenx — APPROVED** [review]

**Review by matsduf — APPROVED** [review]

**Pull request merged**